### ISSUE TITLE
Error in pretty print with line breaks - removes closing tags

### ISSUE THREAD

**Issue by thomasjoscht** [morbac/xmltools#34]
Hi,

I've encountered a problem with function "Pretty print (XML only - with line breaks)". In some situations the endinge tags will be removed which results in an invalid XML. 

The following XML code can be used to reproduce the behaviour:
`<konto xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"><kElementtype>K</kElementtype><uniquename>TESTEXPORT</uniquename><shorttext>Religionswissenschaft</shorttext><defaulttext>TESTEXPORT4_Religionswissenschaft</defaulttext><longtext>Religionswissenschaft</longtext><elementnr></elementnr><comment>&lt;p>Das Modulpaket (au&amp;szlig;ersoziologischer Kompetenzbereich) im Studiengebiet Religionswissenschaft wird in der Modul&amp;uuml;bersicht der fachspezifischen Bestimmungen zum Teilstudiengang "Religionswissenschaft" im Rahmen des Zwei-F&amp;auml;cher-Bachelor-Studiengangs geregelt.&lt;/p></comment><defaultlanguageValue>de</defaultlanguageValue><extId>TEST</extId></konto>`

After Pretty print the ending `</comment>` tag will be missing. Do you have any idea why this happens? Perhaps it lies on JAX-WS escaping mechanismn of strings. It does only escape &lt; but not &gt;. The code above is part of a SOAP request. The comment field contains HTML. Thx

Regards
Thomas

**Comment by morbac** [comment]
Hi,

Thanks for your message. I agree there is an issue when pretty printing following xml:
`<a>"></a>`
I've just commited a fix but I need some more testing before I build a new release. 
Regards
NC

**Comment by thomasjoscht** [comment]
Hi @morbac ,
thx for your investigation. I'm looking forward for a new release.
Regards Thomas

**Comment by morbac** [comment]
Hi,
For your information, the [release 3.0.3.3 available here](https://github.com/morbac/xmltools/releases/tag/3.0.3.3) should fix the issue.
Best Regards

**Comment by thomasjoscht** [comment]
Thx. I've manually changed my Notepad++ extension. The error is fixed. Only the indenting of following attribute is wrong. Its shiftet to the left.

By the way: Do you know whether there is an option to avoid shorten empty tags? like `<elementnr></elementnr>` will become to `<elementnr/>`.

**Comment by morbac** [comment]
Please check the [new version 3.0.4.1](https://github.com/morbac/xmltools/releases/tag/3.0.4.1) which uses a fully rewritten pretty print function. It should fix the indentation errors.

**Comment by thomasjoscht** [comment]
Hey, 

I've tested version 3.0.4.1 with one of my large SOAP-requests. It seems there is still an indention problem with empty elements. There is sometimes a line break missing.

For reproduce try this:
`<abschlusskonto><pruefungen><pruefung><extId>CASE233057_TEST1_AA.8873.10.20182</extId><examinationAttributes/></pruefung></pruefungen><zusatzpruefungen/></abschlusskonto>`

Result is:
```
<abschlusskonto>
	<pruefungen>
		<pruefung>
			<extId>CASE233057_TEST1_AA.8873.10.20182</extId>
			<examinationAttributes/></pruefung>
	</pruefungen>
	<zusatzpruefungen/></abschlusskonto>
```

Regards
Thomas

**Comment by morbac** [comment]
This has been fixed in [3.0.4.2](https://github.com/morbac/xmltools/releases/tag/3.0.4.2)
